### PR TITLE
update number of banks for BIOSANS in MaskBTP

### PR DESCRIPTION
**Pull request by jmborr**
**Description of work**
Update the number of maximum banks for CG3 (BIOSANS) instrument in MaskBTP algorithm

**Purpose of work**
Bugfix [Task 1901 Update Mantid's MaskBTP to handle the Midrange detector](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1901)

**Summary of work**
- Update `bankmax` for CG3 and BIOSANS
- Unit test to make sure MaskBTP works for workspace containing or lacking the latest Midrange detector

**To test:**
In Mantid's workbench, run the following script:
```python
from mantid.simpleapi import LoadEmptyInstrument, MaskBTP
workspace = LoadEmptyInstrument(Filename="CG3_Definition.xml")
MaskBTP(Workspace=workspace, Pixel="1-18,239-256")
```
View the instrument of workspace `workspace`. The top and bottom of every tube should be masked. In the below picture, they will be colored as grey:
Capture

Fixes [Task 1901 Update Mantid's MaskBTP to handle the Midrange detector
](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1901)

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/PythonInterface/test/python/plugins/algorithms/MaskBTPTest.py [hunk 1]
```diff
@@ -198,6 +198,17 @@ def test_biosans_wing_ends(self):
         self.assertEqual(int(32 * 160), len(masked))
         self.checkConsistentMask(wksp, masked)
 
+    def test_biosans_midrange(self):
+        # load latest instrument definition, 104 banks and 4 tubes per bank
+        workspace = LoadEmptyInstrument(Filename="CG3_Definition.xml", OutputWorkspace="CG3csejf234f")
+        masked = MaskBTP(Workspace=workspace, Pixel="1-42")
+        self.assertEqual(42 * 4 * 104, len(masked))
+        # load previous instrument definition, 88 banks and 4 tubes per bank
+        workspace = LoadEmptyInstrument(Filename="CG3_Definition.xml", OutputWorkspace="CG3csejf234f")
```
Comment (backmari) [inline Framework/PythonInterface/test/python/plugins/algorithms/MaskBTPTest.py:207]:
```suggestion
        workspace = LoadEmptyInstrument(Filename="CG3_Definition_2019_2023.xml", OutputWorkspace="CG3csejf234f")
```

File: Framework/PythonInterface/test/python/plugins/algorithms/MaskBTPTest.py [hunk 2]
```diff
@@ -198,6 +198,17 @@ def test_biosans_wing_ends(self):
...
+        masked = MaskBTP(Workspace=workspace, Pixel="1-42")
+        self.assertEqual(42 * 4 * 104, len(masked))
```
Comment (backmari) [inline Framework/PythonInterface/test/python/plugins/algorithms/MaskBTPTest.py:209]:
```suggestion
        self.assertEqual(42 * 4 * 88, len(masked))
```

**Comment by backmari** [comment]
I think the second part of the test was supposed to be for the previous file "CG3_Definition_2019_2023.xml"?

**Comment by backmari** [comment]
The manual test works as expected:
image

**Review by backmari — DISMISSED** [review]

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
Needs release notes

**Comment by jmborr** [comment]
> Needs release notes
included now

**Review by backmari — APPROVED** [review]

**Review by AndreiSavici — APPROVED** [review]

**Pull request merged**